jfrog-cli/artifactory/utils/golang/cmd.go

```diff
@@ -105,13 +107,70 @@ func RunGo(goArg string) error {
 	return utils.RunCmdWithOutputParser(goCmd, &protocolRegExp, &notFoundRegExp)
 }
 
-// Using go mod download command to download all the dependencies before publishing to Artifactory
-func DownloadDependenciesDirectly() error {
+// Using go mod download {dependency} command to download the dependency
+func DownloadDependency(dependencyName string) error {
 	goCmd, err := NewCmd()
 	if err != nil {
 		return err
 	}
 
-	goCmd.Command = []string{"mod", "download"}
+	goCmd.Command = []string{"mod", "download", "-json", dependencyName}
 	return utils.RunCmd(goCmd)
 }
+
+// Runs go mod graph command and returns slice of the dependencies
+func GetDependenciesGraph() (map[string]bool, error) {
+	log.Info("Running go mod graph command")
+	goCmd, err := NewCmd()
+	if err != nil {
+		return nil, err
+	}
+	goCmd.Command = []string{"mod", "graph"}
+	output, err := utils.RunCmdOutput(goCmd)
+	log.Debug(string(output))
+	return outputToMap(string(output)), errorutils.CheckError(err)
+}
+
+func outputToMap(output string) map[string]bool {
+	lineOutput := strings.Split(output, "\n")
+	var result []string
+	mapOfDeps := map[string]bool{}
+	for _, line := range lineOutput {
+
+		splitLine := strings.Split(line, " ")
+		if len(splitLine) == 2 {
+			mapOfDeps[splitLine[1]] = true
+			result = append(result, splitLine[1])
+		}
+	}
+	return mapOfDeps
+}
+
+// Using go mod download command to download all the dependencies before publishing to Artifactory
+func RunGoModTidy() error {
+	log.Info("Running go mod tidy command")
+	goCmd, err := NewCmd()
+	if err != nil {
+		return err
+	}
+
+	goCmd.Command = []string{"mod", "tidy"}
+	_, err = utils.RunCmdOutput(goCmd)
+	return err
+}
+
+// Returns the root dir where the go.mod located.
+func GetRootDir() (string, error) {
```

Comment:
Looks like after every invocation of this function, *strings.TrimSpace* is used on the result.
Can we move the *strings.TrimSpace* into this function?

Reply:
Moved the strings.TrimSpace into GetRootDir.